bugwarrior/services/clickup.py

```diff
@@ -0,0 +1,150 @@
+            page += 1
+
+
+class ClickupIssue(Issue):
+    ID = "clickupid"
+    DESCRIPTION = "clickupdescription"
+    STATUS = "clickupstatus"
+    UPDATED_AT = "clickupupdated"
+    CREATOR = "clickupcreator"
+    URL = "clickupurl"
+    LIST_NAME = "clickuplistname"
+    PROJECT = "clickupproject"
+    FOLDER = "clickupfolder"
+    SPACE = "clickupspace"
+
+    UDAS = {
+        ID: {"type": "string", "label": "Clickup id"},
+        DESCRIPTION: {"type": "string", "label": "Clickup Description"},
+        STATUS: {"type": "string", "label": "Clickup Status Title"},
+        UPDATED_AT: {"type": "date", "label": "Clickup Updated"},
+        CREATOR: {"type": "string", "label": "Clickup Creator"},
+        URL: {"type": "string", "label": "Clickup URL"},
+        LIST_NAME: {"type": "string", "label": "Clickup List name"},
+        PROJECT: {"type": "string", "label": "Clickup Project id"},
+        FOLDER: {"type": "string", "label": "Clickup Folder id"},
+        SPACE: {"type": "string", "label": "Clickup Space id"},
+    }
+    UNIQUE_KEY = [ID]
+
+    PRIORITY_MAP = {"urgent": "H", "high": "M", "normal": "L", "low": "", None: ""}
+
+    def to_taskwarrior(self):
+        self.title = self.record["name"]
+
+        if not self.record["project"]["hidden"]:
+            project = self.record["project"]["name"]
+        else:
+            project = None
+
+        return {
+            "project": project,
+            "priority": self.record["priority"],
+            "annotations": self.extra.get("annotations", []),
+            "due": self.parse_timestamp(self.record["due_date"]),
+            "entry": self.parse_timestamp(self.record["date_created"]),
+            self.ID: self.record["id"],
+            self.DESCRIPTION: self.record["description"],
+            self.STATUS: self.record["status"]["status"],
+            self.UPDATED_AT: self.parse_timestamp(self.record["date_updated"]),
+            self.CREATOR: self.record["creator"]["username"],
+            self.URL: self.record["url"],
+            self.LIST_NAME: self.record["list"]["name"],
+            self.PROJECT: self.record["project"]["id"],
+            self.FOLDER: self.record["folder"]["id"],
+            self.SPACE: self.record["space"]["id"],
+        }
+
+    def get_default_description(self):
+        return self.build_default_description(title=self.title, url=self.record["url"])
+
+    @staticmethod
+    def parse_timestamp(
+        milliseconds_unix: Optional[str],
+    ) -> Optional[datetime.datetime]:
+        if milliseconds_unix is None:
+            return None
+
+        seconds_unix = float(milliseconds_unix) / 1e3
+        return datetime.datetime.fromtimestamp(seconds_unix)
+
+
+class ClickupService(Service):
+    API_VERSION = 1.0
+    ISSUE_CLASS = ClickupIssue
+    CONFIG_SCHEMA = ClickupConfig
+
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+        self.client = ClickupClient(token=self.config.token)
```

Comment:
This allows this value to be retrieved from an [oracle](https://bugwarrior.readthedocs.io/en/latest/common_configuration.html#secret-management).

```suggestion
        self.client = ClickupClient(token=self.get_secret('token'))
```